static/js/src/ua-payment-modal.js

```diff
@@ -316,35 +323,49 @@ function applyTotals() {
     country: country,
   };
 
-  postCustomerInfoForPurchasePreview(
-    currentTransaction.accountId,
-    addressObject,
-    taxObject
-  )
-    .then(() => {
-      if (currentTransaction.type === "purchase") {
-        postPurchasePreviewData(
-          currentTransaction.accountId,
-          currentTransaction.products,
-          currentTransaction.previousPurchaseId
-        ).then((data) => {
-          purchasePreview = data;
-          modal.classList.remove("is-processing");
-          setOrderTotals(country, vatApplicable, purchasePreview, modal);
-        });
-      } else if (currentTransaction.type === "renewal") {
-        postRenewalPreviewData(currentTransaction.transactionId).then(
-          (data) => {
+  if (currentTransaction.accountId) {
+    postCustomerInfoForPurchasePreview(
+      currentTransaction.accountId,
+      addressObject,
+      taxObject
+    )
+      .then(() => {
+        if (currentTransaction.type === "purchase") {
+          postPurchasePreviewData(
+            currentTransaction.accountId,
+            currentTransaction.products,
+            currentTransaction.previousPurchaseId
+          ).then((data) => {
             purchasePreview = data;
             modal.classList.remove("is-processing");
             setOrderTotals(country, vatApplicable, purchasePreview, modal);
-          }
-        );
-      }
-    })
-    .catch((error) => {
-      console.error(error);
-    });
+          });
+        } else if (currentTransaction.type === "renewal") {
+          postRenewalPreviewData(currentTransaction.transactionId).then(
+            (data) => {
+              purchasePreview = data;
+              modal.classList.remove("is-processing");
+              setOrderTotals(country, vatApplicable, purchasePreview, modal);
+            }
+          );
+        }
+      })
+      .catch((error) => {
+        console.error(error);
+      });
+  } else {
+    const purchaseTotals = {
+      total: currentTransaction.subtotal,
+    };
+    guestPurchase = true;
+    modal.classList.remove("is-processing");
+
+    // set the "Country" and vatApplicable parameters
```

Comment:
In which scenario would this happen (not having an account ID)?

Reply:
Yeah, this is a guest purchase thing only - because at this point I don't have enough information to send to the purchase preview endpoint and get an actual VAT amount back, we just use the unit price * number of machines from the form to give a subtotal.